src/main/java/zmaster587/advancedRocketry/tile/multiblock/machine/TileChemicalReactor.java

```diff
@@ -54,23 +57,18 @@ public void onInventoryUpdated() {
 		//If we are already processing something don't bother
 		IRecipe recipe;
 		boolean flag = false;
+
 		if(getOutputs() == null && (recipe = getRecipe(getMachineRecipeList())) != null && canProcessRecipe(recipe))
 		{
 			if(!recipe.getOutput().isEmpty()) {
-			NBTTagList list = recipe.getOutput().get(0).getEnchantmentTagList();
-
-			/*
-				for( int i = 0 ; i < list.tagCount(); i++ ) {
+				NBTTagList list = recipe.getOutput().get(0).getEnchantmentTagList();
+				for(int i = 0 ; i < list.tagCount(); i++) {
 					NBTTagCompound tag = (NBTTagCompound)list.get(i);
 					if(tag.getInteger("id") == Enchantment.getEnchantmentID(AdvancedRocketryAPI.enchantmentSpaceProtection) ) {
-
 						flag = true;
 						break;
 					}
 				}
```

Comment:
Cache enchantment ID before loop. Enchantment ID won't change during loop. Also you can use EnchantmentHelper.getEnchantmentLevel(enchantment: Enchantment, stack: ItemStack): Int instead probably but it's up to you

Reply:
Done